### PR TITLE
Redis lua script for atomic operations on redis

### PR DESCRIPTION
**Pull request by matsboman**
# Resolves

https://jira.rax.io/browse/CERES-523

# What

Add lua script for atomic operations on redis

# How

Added script bean

## How to test

Execute JUnit test

### CODE REVIEW

File: src/main/resources/check-partition-job.lua [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--lua
+local old_time=redis.call('get', KEYS[1])
```
Comment (shabd67) [inline src/main/resources/check-partition-job.lua:2]: in place of this Lua script, can we use SETNX method of Redis: 
https://redis.io/commands/setnx

Comment (matsboman) [reply src/main/resources/check-partition-job.lua:2]: "Design pattern: Locking with SETNX

Please note that:

The following pattern is discouraged in favor of the Redlock algorithm which is only a bit more complex to implement, but offers better guarantees and is fault tolerant..."

I think LUA script is safer and according to my testing so far no non-atomic instances occur with LUA script.

Comment (shabd67) [reply src/main/resources/check-partition-job.lua:2]: can we have steps to test it in docs-internal repo ?

Comment (matsboman) [reply src/main/resources/check-partition-job.lua:2]: That will be in the next PR. This PR provides JUnit only

**Review by shabd67 — APPROVED** [review]

**Review by rohitsngh27 — APPROVED** [review]
LGTM

**Pull request merged**